ruby/tests/encode_decode_test.rb

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/ruby
+
+# generated_code.rb is in the same directory as this test.
+$LOAD_PATH.unshift(File.expand_path(File.dirname(__FILE__)))
+
+require 'generated_code_pb'
+require 'test/unit'
+
+def hex2bin(s)
+  s.scan(/../).map { |x| x.hex.chr }.join
+end
+
+class EncodeDecodeTest < Test::Unit::TestCase
+  def test_discard_unknown
+    # Test discard unknown in message.
+    from = hex2bin('F80601')
```

Comment:
The hex strings aren't readable. Could you create the proto by using a different message which has that field defined? Then you could serialize it.